feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,745 @@
+	controllerCardType    = oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_CONTROLLER_CARD
+	cpuHighUtilization    = 80
+	memoryHighUtilization = 80
+)
+
+// coreFileCheck function is used to check if cores are found on the DUT.
+func coreFileCheck(t *testing.T, dut *ondatra.DUTDevice, gnoiClient raw.GNOI, sysConfigTime uint64, retry bool) {
+	t.Helper()
+	t.Log("Checking for core files on DUT")
+
+	// vendorCoreFilePath and vendorCoreProcName should be provided to fetch core file on dut.
+	if _, ok := vendorCoreFilePath[dut.Vendor()]; !ok {
+		t.Fatalf("Please add support for vendor %v in var vendorCoreFilePath ", dut.Vendor())
+	}
+	if _, ok := vendorCoreFileNamePattern[dut.Vendor()]; !ok {
+		t.Fatalf("Please add support for vendor %v in var vendorCoreFileNamePattern.", dut.Vendor())
+	}
+
+	in := &fpb.StatRequest{
+		Path: vendorCoreFilePath[dut.Vendor()],
+	}
+	validResponse, err := gnoiClient.File().Stat(context.Background(), in)
+	if err != nil {
+		if retry {
+			t.Logf("Retry GNOI request to check %v for core files on DUT", vendorCoreFilePath[dut.Vendor()])
+			validResponse, err = gnoiClient.File().Stat(context.Background(), in)
+		}
+	}
+	if err != nil {
+		t.Fatalf("Unable to stat path %v for core files on DUT, %v", vendorCoreFilePath[dut.Vendor()], err)
+	}
+	// Check cores creation time is greater than test start time.
+	for _, fileStatsInfo := range validResponse.GetStats() {
+		if fileStatsInfo.GetLastModified() > sysConfigTime {
+			coreFileName := fileStatsInfo.GetPath()
+			r := vendorCoreFileNamePattern[dut.Vendor()]
+			if r.MatchString(coreFileName) {
+				t.Errorf("Found core %v on DUT.", coreFileName)
+			}
+		}
+	}
+}
+
+func sortPorts(ports []*ondatra.Port) []*ondatra.Port {
+	sort.Slice(ports, func(i, j int) bool {
+		idi, idj := ports[i].ID(), ports[j].ID()
+		li, lj := len(idi), len(idj)
+		if li == lj {
+			return idi < idj
+		}
+		return li < lj // "port2" < "port10"
+	})
+	return ports
+}
+
+func contains(list []string, target string) bool {
+	for _, item := range list {
+		if item == target {
+			return true
+		}
+	}
+	return false
+}
+
+func removeElement(list []string, element string) []string {
+	for i := 0; i < len(list); i++ {
+		if list[i] == element {
+			list = append(list[:i], list[i+1:]...)
+			i-- // Adjust index to account for removed element
+		}
+	}
+	return list
+}
+
+func TestGetComponentNames(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	timestamp = uint64(time.Now().UTC().Unix())
+
+	controllerCards = components.FindComponentsByType(t, dut, controllerCardType)
+	_, rpActive = components.FindStandbyRP(t, dut, controllerCards)
+	lineCards = components.FindComponentsByType(t, dut, lineCardType)
+	fabricCards = components.FindComponentsByType(t, dut, fabricCardType)
+	cpuCards = components.FindComponentsByType(t, dut, cpuType)
+
+	checkComponents = append(controllerCards, lineCards...)
+	checkComponents = append(checkComponents, fabricCards...)
+
```

Comment:
Delete blank line

Reply:
Thanks for checking. Fixed it. Deleted extra blank line.
```go
	cpuCards = components.FindComponentsByType(t, dut, cpuType)
	checkComponents = append(controllerCards, lineCards...)
	checkComponents = append(checkComponents, fabricCards...)
}
```